src/live.js

```diff
@@ -28,65 +28,114 @@ function decodeHTMLEntities(text) {
+    var regex = /StreamTitle\='(.+?) - (.+?)';/;
+    var matches = metadata.match(regex);
+    if (matches) {
+        return {
+            artist: decodeHTMLEntities(matches[1]),
+            track: decodeHTMLEntities(matches[2])
+        }
+    } else {
+        return null;
     }
-  } else {
-    return null;
-  }
 }
 
 Object.keys(module.exports).forEach((key) => {
-  module.exports[key].on('newListener', (event) => {
-    if(event !== 'song') return;
-    if(module.exports[key].listenerCount('song') == 0) {
-      icy.get(URLS[key], (res) => {
-        res.on("metadata", (metadata) => {
-          metadata = parseMetadata(metadata.toString());
-          if(metadata != null && metadata.artist != '414 Music') {
-            musicbrainz.artist(metadata.artist).then((artistObj) => {
-              return musicbrainz.track(metadata.track, artistObj).then((track) => {
-                module.exports[key].emit('song', {
-                  "artist": artistObj.name,
-                  "track": track.title,
-                  "album": track.releases[0].title,
-                  "playedAt": new Date()
+    module.exports[key].on('newListener', (event) => {
+        if (event !== 'song') return;
+        if (module.exports[key].listenerCount('song') == 0) {
+            icy.get(URLS[key], (res) => {
+                res.on("metadata", (metadata) => {
+                    metadata = parseMetadata(metadata.toString());
+                    if (metadata != null && metadata.artist != '414 Music') {
+                        getMusicbrainz(metadata).then((obj) => {
+                            module.exports[key].emit('song', obj);
+                        })
+                    }
                 });
-              })
-            }).catch((e) => {
-              module.exports[key].emit('song', {
-                "artist": metadata.artist,
-                "track": metadata.track,
-                "album": null,
-                "playedAt": new Date()
-              });
-            });
-          }
-        })
-        res.pipe(devnull());
-        PIPES[key] = res;
-      })
-    }
-  })
-  module.exports[key].on('removeListener', () => {
-    if(module.exports.listenerCount('song') == 1) {
-      if(PIPES[key]) {
-        PIPES[key].unpipe(devnull());
-        PIPES[key] = null;
-      }
-    }
-  });
+                res.pipe(devnull());
+                PIPES[key] = res;
+            })
+        }
+    })
+    module.exports[key].on('removeListener', () => {
+        if (module.exports.listenerCount('song') == 1) {
+            if (PIPES[key]) {
+                PIPES[key].unpipe(devnull());
+                PIPES[key] = null;
+            }
+        }
+    });
 })
 
+
+function getMusicbrainz(metadata) {
+    let defaultObject = {
+        "artist": metadata.artist,
+        "track": metadata.track,
+        "album": null,
+        "playedAt": new Date(),
+        "coverArt": null,
+        "musicBrainz": {
```

Comment:
I returned the music brainz resp incase in the future I want to use this and not have to PR this lib.